### ISSUE TITLE
Running Storyshots in a clean HTML app fails

### ISSUE THREAD

**Issue by ziyafenn** [storybookjs/storybook#19175]
I installed Storybook and Storyshots addon on a clean HTML project with Jest. Followed the guidelines on how to initialize Storyshots, but every time i run the jest command to initialize the Storyshots, i get the following error: 

``` FAIL  ./storybook.test.js
  ● Test suite failed to run

    Couldn't find an appropriate framework loader -- do you need to set the `framework` option?

      2 |
      3 | import initStoryshots from '@storybook/addon-storyshots';
    > 4 | initStoryshots();
        | ^

      at Object.loadFramework [as default] (node_modules/@storybook/addon-storyshots/dist/ts3.9/frameworks/frameworkLoader.js:24:15)
      at testStorySnapshots (node_modules/@storybook/addon-storyshots/dist/ts3.9/api/index.js:28:94)
      at Object.<anonymous> (storybook.test.js:4:1)
```

After explicitly setting framework property within `initStoryshots` method to `HTML`, i get the following error:

```
 FAIL  ./storybook.test.js
  ● Test suite failed to run

    TypeError: Cannot set properties of undefined (setting 'STORYBOOK_ENV')

      at Object.<anonymous> (node_modules/@storybook/html/dist/cjs/client/preview/globals.js:8:28)
      at Object.<anonymous> (node_modules/@storybook/html/dist/cjs/client/preview/index.js:12:1)
```

Storybook is installed using WebPack5 builder but also tested on version 4

**Comment by ndelangen** [comment]
@ziyafenn We have a bunch of storyshots examples/test files in our repo:
https://github.com/storybookjs/storybook/blob/af0578d99ca3972d01700d7b4aa8c7f2de49ab5e/code/addons/storyshots-core/stories/storyshot.async.test.js#L1-L41

Perhaps these help you?

Is suspect this issue is actually fixed in 7.0 beta.

**Comment by loganzartman** [comment]
Possible solution: https://github.com/storybookjs/storybook/issues/23752#issuecomment-1722070821
